nixos/modules/system/boot/stage-1-init.sh

```diff
@@ -423,8 +423,15 @@ mountFS() {
 
     # For bind mounts, busybox has a tendency to ignore options, which can be a
     # security issue (e.g. "nosuid"). Remounting the partition seems to fix the
-    # issue.
-    mount "/mnt-root$mountPoint" -o "remount,$optionsPrefixed"
+    # issue. This should only be done for bind and rbind mountpoints because other
+    # filesystems may have limitations or may not support remount.
+    local IFS=,
+    for opt in $options; do
+        if [ "$opt" = bind ] || [ "$opt" = rbind ]; then
+            mount "/mnt-root$mountPoint" -o "remount,$optionsPrefixed"
+            return
```

Comment:
There's no case in which mount options are specified multiple times (or in multiple ways, e.g. both `bind` and `rbind`), correct? Otherwise I worry an early return would be bad.

Reply:
I don't think both  The desired logic is `options.contains("bind") || options.contains("rbind")` but this a bash so exiting a for-loop early seems like the best solution.